File: tests/fixtures/python/ThrowVarargsConstructor.py
from __future__ import annotations


class ThrowVarargsConstructor:

    def check(self, found: str, position: int, *expected: str) -> None:
        raise UnexpectedTokenException(found, position, *expected)


class UnexpectedTokenException(Exception):

    def __init__(self, found: str, position: int, *expected: str) -> None:
        super().__init__(found)

File: tests/fixtures/java/ThrowVarargsConstructor.java
public class ThrowVarargsConstructor {
    public void check(String found, int position, String... expected) {
        throw new UnexpectedTokenException(found, position, expected);
    }
}

class UnexpectedTokenException extends RuntimeException {
    UnexpectedTokenException(String found, int position, String... expected) {
        super(found);
    }
}
